### DISCUSSION TITLE
Husk limb causes crash if removed while targeted by ranged/monster attack.

### DISCUSSION THREAD

**Discussion by Draconis13** [FakeFishGames/Barotrauma#14819] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

So this is a bit esoteric but can occur in normal play in certain obscure circumstances. Husk limbs are supposed to be Target priority 0 but attacks especially ranged attacks can still target them, in certain circumstances such as poison removing husk and with it the tentacle or removing a genesplicer or otherwise removing a vanilla or modded husk limb on a character while that limb is being targeted will crash the game with an error related to a AI targeting null limb.

Most of the time this will be extremely corner case such as instantly killed while having a husktentacles/mudraptorbeak or removing a genesplicer while being shot at by a spineling but it is still a significant and repeatable crash to desktop.

One of the mod makers I was talking to was commonly running into this issue  and did a work around by having their custom husklimbs be always present but Alpha 0 so that the effect of the limb and its alpha state were only present when needed but the target crash bug would not occur as it was always present.

### Reproduction steps

Have a character with one or more husk limbs be targeted by AI, especially ranged weapon attacks, the husk limb even though priority 0 will still occasionally be targeted, remove husk limb via state such as removal of gene splicer or modded clothing/item and game/server crashes. 

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.5.9.2 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report!

I gave this a test, but unfortunately could not reproduce the issue. The monsters seem to stop targeting the limbs as soon as they disappear.

Any chance you could share the crash report?

**Comment by Draconis13** [comment]
I can try to replicate it again, Did you test it with hostile NPC AI holding guns or other ranged weapons? That seemed to be the primary issue  as it gave a nasty null limb error, I also don't know if it is because limbpriority="0" is causing the issue to begin with

**Reply by Regalis11** [reply]
I only tested with this monsters, but I'll give it another shot with human enemies in case the issue is in the human AI. 👍
